From 3aed924739420c15341f377ebd60a7918111163f Mon Sep 17 00:00:00 2001
From: kunleiz <kunleiz@codeaurora.org>
Date: Mon, 18 Feb 2019 16:46:34 +0800
Subject: [PATCH 1/2] policy_hal: do not check MODE_IN_COMMUNICATION for
 voip_rx flag

Some VoIP clients may choose to open the streams as soon as the phone
state is set to MODE_RINGTONE to allow lower start latency when the call
is connected. This patch removes the unnecessary check conditions for
MODE_IN_COMMUNICATION since the stream type is sufficient indicator for
setting the flags for VoIP calls.

Bug: 62782025
Commit: 72ad4b91d6ece44b7967080c44695227c6789147
Change-Id: Idc7e1b6f33921580c0937dee71b2b2105cbd31eb

CRs-Fixed: 2399711
Change-Id: I68b9204210a857cf78e9b7467671cd8013a3d789
---
 policy_hal/AudioPolicyManager.cpp | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/policy_hal/AudioPolicyManager.cpp b/policy_hal/AudioPolicyManager.cpp
index b15e96e..490eeb4 100644
--- a/policy_hal/AudioPolicyManager.cpp
+++ b/policy_hal/AudioPolicyManager.cpp
@@ -1493,8 +1493,7 @@ audio_io_handle_t AudioPolicyManagerCustom::getOutputForDevice(
        }
 
 #ifdef COMPRESS_VOIP_ENABLED
-    if ((mEngine->getPhoneState() == AUDIO_MODE_IN_COMMUNICATION) &&
-        (stream == AUDIO_STREAM_VOICE_CALL) &&
+    if (stream == AUDIO_STREAM_VOICE_CALL &&
         audio_is_linear_pcm(config->format)) {
         // let voice stream to go with primary output by default
         // in case direct voip is bypassed
@@ -1542,8 +1541,7 @@ audio_io_handle_t AudioPolicyManagerCustom::getOutputForDevice(
             *flags = (audio_output_flags_t)(AUDIO_OUTPUT_FLAG_FAST|AUDIO_OUTPUT_FLAG_PRIMARY);
         }
 #else
-    if (mEngine->getPhoneState() == AUDIO_MODE_IN_COMMUNICATION &&
-        stream == AUDIO_STREAM_VOICE_CALL &&
+    if (stream == AUDIO_STREAM_VOICE_CALL &&
         audio_is_linear_pcm(config->format)) {
         //check if VoIP output is not opened already
         bool voip_pcm_already_in_use = false;
-- 
2.17.1

